src/main/java/org/broadinstitute/dsm/route/FilterRoute.java

```diff
@@ -296,8 +300,32 @@ public Object processRequest(Request request, Response response, String userId)
         }
     }
 
-    public static void addParticipantDataFilters(Map<String, String> queryConditions,
-                                                 Filter filter, String fieldName, List<ParticipantDataDto> allParticipantData) {
+    public static void addParticipantDataConditionsToQuery(Map<String, Integer> allIdsForParticipantDataFiltering, Map<String, String> queryConditions, int filtersLength) {
+        StringBuilder newCondition = new StringBuilder(ElasticSearchUtil.AND);
+        int i = 0;
+        if (allIdsForParticipantDataFiltering.isEmpty()) {
+            queryConditions.put(ElasticSearchUtil.ES, ElasticSearchUtil.BY_PROFILE_GUID + ElasticSearchUtil.EMPTY);
+        } else {
+            for (Map.Entry<String, Integer> entry: allIdsForParticipantDataFiltering.entrySet()) {
+                if (entry.getValue() != filtersLength) {
+                    continue;
+                }
+                if (i == 0) {
+                    newCondition.append(ParticipantUtil.isGuid(entry.getKey()) ? ElasticSearchUtil.BY_PROFILE_GUID + entry.getKey() : ElasticSearchUtil.BY_PROFILE_LEGACY_ALTPID + entry.getKey());
+                } else {
+                    newCondition.append(ParticipantUtil.isGuid(entry.getKey()) ? ElasticSearchUtil.BY_GUIDS + entry.getKey() : ElasticSearchUtil.BY_LEGACY_ALTPIDS + entry.getKey());
+                }
+                i++;
+            }
+            newCondition.append(ElasticSearchUtil.CLOSING_PARENTHESIS);
+            String esCondition = queryConditions.get(ElasticSearchUtil.ES);
+            esCondition += newCondition.toString();
+            queryConditions.put(ElasticSearchUtil.ES, esCondition);
```

Comment:
code from line 321 to 324 could be minimized using Map's `computeIfPresent`,
`queryConditions.computeIfPresent(ElasticSearchUtil.ES, (key, value) -> value += newCondition.toString())`,
except that `else` block's content could be extracted as a method with name something like `createParticipantIdsQuery` or something like that, also `i` and `newCondition` is used only in `else` block so they can be easily moved to extracted method.